framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -628,10 +622,18 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 	private async _deleteBlockHandler(data?: Record<string, unknown>) {
 		const { blockHeader: receivedBlock } = data as unknown as Data;
 
-		const newBlockHeader = chain.BlockHeader.fromJSON(receivedBlock).toObject();
+		const deletedBlockHeader = chain.BlockHeader.fromJSON(receivedBlock).toObject();
+
+		// Delete ccmEvents for the height of blockHeader
+		const crossChainMessages = await this._chainConnectorStore.getCrossChainMessages();
+		const indexForCCMEvents = crossChainMessages.findIndex(
+			ccm => ccm.height >= deletedBlockHeader.height,
+		);
+		crossChainMessages.splice(indexForCCMEvents, 1);
```

Comment:
A clarification: here we are removing one element at index `indexForCCMEvents`? Why not all elements from that index on?

Reply:
we want to delete only the element which was corresponding to the deleted block height